exchange/howto/run-seednode.adoc

```diff
@@ -4,108 +4,306 @@
+
+----
+[Unit]
+Description=Bitcoind
+After=network.target
+
+[Service]
+ExecStart=bitcoind -daemon \
+                   -printtoconsole \
+                   -nodebuglogfile \
+                   -pid=/home/bitcoind/.bitcoin/bitcoind.pid
+ExecStop=/home/bitcoind/bitcoin-cli stop
+
+Type=forking
+PIDFile=/home/bitcoind/.bitcoin/bitcoind.pid
+Restart=on-failure
+
+User=bitcoind
+Group=bitcoind
+
+PrivateTmp=true
+ProtectSystem=full
+NoNewPrivileges=true
+PrivateDevices=true
+MemoryDenyWriteExecute=true
+
+[Install]
+WantedBy=multi-user.target
+----
+
+Create a file `/home/bitcoind/.bitcoin/bitcoin.conf` that contains
+
+----
+server=1
+txindex=1
+dbcache=1000
+maxconnections=800
+timeout=30000
+listen=0
+rpcallowip=127.0.0.1
+rpcuser=YOUR_USER_NAME
+rpcpassword=YOUR_PW
+blocknotify=sh ~/.bitcoin/blocknotify.sh %s
+----
+
+and another file `/home/bitcoind/.bitcoin/blocknotify.sh` that contains
+
+  #!/bin/sh
+  echo $1 | nc -w 1 127.0.0.1 5120
+
+and make it executable (`chmod +x /home/bitcoind/.bitcoin/blocknotify.sh`).
+
+Finally, enable and start the service
+
+  systemctl enable bitcoind.service
+  systemctl start bitcoind.service
+
+and observe the logs
+
+  journalctl --unit bitcoind --follow
+
+an check if anything works as expected.
+
+=== Bisq Seed Node
+
+Start by getting OpenJDK 10 up and running.
+
+For getting the Bisq binaries, we recommend cloning the Bisq Git repository and compiling the code on your server. This way, you have precise control over what version you want to deploy. Furthermore, updating is very simple, just pull the changes, recompile and restart your service.
+
+Furthermore, we recommend creating a user `bisq` in group `bisq` for service hardening reasons and using the `bisq`-users home directory to:
+
+  useradd -r -m bisq
+  cd /home/bisq
+  git clone git@github.com:bisq-network/bisq.git
+  cd bisq
+  ./gradlew build -x test
+
+Create a systemd service file `bisq-seednode.service` (or copy the one shipped with bisq `$bisqdir/seednode/bisq-seednode.service`) in the systemd service path or your operating system (something like `/usr/lib/systemd/system/`) and adapt it to your needs.
+
+In the end, your file should look something like
+
+----
+[Unit]
+Description=Bisq Seed Node
+After=network.target
+
+[Service]
+Environment="JAVA_OPTS=-Xms512M -Xmx2000M -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=6969 -Dcom.sun.management.jmxremote.rmi.port=6969 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"
+ExecStart=/home/bisq/bisq/bisq-seednode --appName=bisq-seednode --nodePort=8000 --userDataDir=/home/bisq/ --maxConnections=50 --fullDaoNode=true --rpcUser=YOUR_USER_NAME --rpcPassword=YOUR_PW --rpcPort=8332 --rpcBlockNotificationPort=5120
```

Comment:
Not sure if maxConnections 50 is too high. Typically we use 30.

Reply:
Back when I ran 2 seed nodes, I had it set to 50 with only 800MB of Xmx and it worked without running out of memory. How about changing it to 30 and then adjust once we have a continuous heap space reading of the seed nodes?

Reply:
Yes I think that makes sense. I am willing to be a guinea pig and optimize these settings on one seed once I add monitoring to it.